shared/build.gradle.kts

```diff
@@ -93,3 +120,61 @@ android {
 tasks.check {
     dependsOn(detekt)
 }
+
+buildkonfig {
+    packageName = "co.nimblehq.blisskmmic"
+
+    defaultConfigs {
+        buildConfigField(
+            STRING,
+            "CLIENT_ID",
+            BuildKonfig.CLIENT_ID
+        )
+        buildConfigField(
+            STRING,
+            "CLIENT_SECRET",
+            BuildKonfig.CLIENT_SECRET
+        )
+        buildConfigField(
+            STRING,
+            "BASE_URL",
+            BuildKonfig.STAGING_BASE_URL
+        )
+    }
+
+    defaultConfigs("production") {
+        buildConfigField(
+            STRING,
+            "CLIENT_ID",
+            BuildKonfig.CLIENT_ID
+        )
+        buildConfigField(
+            STRING,
+            "CLIENT_SECRET",
+            BuildKonfig.CLIENT_SECRET
+        )
+        buildConfigField(
+            STRING,
+            "BASE_URL",
+            BuildKonfig.PRODUCTION_BASE_URL
+        )
+    }
+
+    defaultConfigs("staging") {
+        buildConfigField(
+            STRING,
+            "CLIENT_ID",
+            BuildKonfig.CLIENT_ID_STAGING
+        )
+        buildConfigField(
+            STRING,
+            "CLIENT_SECRET",
+            BuildKonfig.CLIENT_SECRET_STAGING
+        )
+        buildConfigField(
+            STRING,
+            "BASE_URL",
+            BuildKonfig.STAGING_BASE_URL
+        )
+    }
```

Comment:
@blyscuit Sorry for recalling you on a merged PR, I think there is still something wrong here. We need to define a properties file for [BuildKonfig](https://github.com/yshrsmz/BuildKonfig) and read them from the properties file instead of a `BuildKonfig` class. The logic here is BuildKonfig lib helps us to create `BuildKonfig` class file to use in runtime from a properties config, so `BuildKonfig` class is the result, not the source of this gradle config.